workflow/controller/workflowpod.go

```diff
@@ -128,6 +128,14 @@ func (woc *wfOperationCtx) createWorkflowPod(nodeName string, mainCtr apiv1.Cont
 		pod.ObjectMeta.Labels[common.LabelKeyControllerInstanceID] = woc.controller.Config.InstanceID
 	}
 
+	err := woc.addArchiveLocation(pod, tmpl)
+	if err != nil {
+		return nil, err
+	}
+
+	volumes, volumeMounts := woc.createSecretVolumes(tmpl)
```

Comment:
Can we move secret volume addition into addVolumeReferences(), where the other volume manipulation currently happens? Then all volume manipulations can happen in the same place. Inside addVolumeReferences(), we can iterate initContainers and containers, and add the volume mounts if the container name is `init` or `wait`.